src/models/neural_networks/linear_network.py

```diff
@@ -81,14 +96,27 @@ def init_weights(self):
         # see: Initializing the biases
         nn.init.constant_(self.final_dense.bias.data, 0)
 
-    def forward(self, x, pred_month=None):
-        # flatten
+    def forward(self, x, pred_month=None, current=None):
+        # flatten the final 2 dimensions (time / feature)
         x = x.contiguous().view(x.shape[0], -1)
+
+        # concatenate the one_hot_month matrix onto X
         if self.include_pred_month:
+            # flatten the array
+            pred_month = pred_month.contiguous().view(x.shape[0], -1)
             x = torch.cat((x, pred_month), dim=-1)
+
+        # concatenate the non-target variables onto X
+        if self.experiment == 'nowcast':
+            assert current is not None
+            current = current.contiguous().view(x.shape[0], -1)
```

Comment:
Shouldn't `current` have shape `[batch_size, dimensions]` anyway? Is this reshaping necessary?

Needs to be changed in the RNN too if not - I just copied the code from here over to there for the tests to pass.

Reply:
yes most likley - i've just copied that down from before for `pred_months` etc. but actially it's not necessary